core/dbt/parser/manifest.py

```diff
@@ -367,82 +318,132 @@ def read_parse_results(self) -> Optional[ParseResult]:
         if os.path.exists(path):
             try:
                 with open(path, 'rb') as fp:
-                    result: ParseResult = pickle.load(fp)
+                    manifest: Manifest = pickle.load(fp)
                 # keep this check inside the try/except in case something about
                 # the file has changed in weird ways, perhaps due to being a
                 # different version of dbt
-                if self.matching_parse_results(result):
-                    return result
+                if self.matching_parse_results(manifest):
+                    return manifest
             except Exception as exc:
                 logger.debug(
                     'Failed to load parsed file from disk at {}: {}'
                     .format(path, exc),
                     exc_info=True
                 )
-
         return None
 
-    def process_manifest(self, manifest: Manifest):
+    # This find the sources, refs, and docs and resolves them
+    # for nodes and exposures
+    def process_manifest(self):
         project_name = self.root_project.project_name
-        process_sources(manifest, project_name)
-        process_refs(manifest, project_name)
-        process_docs(manifest, self.root_project)
-
-    def create_manifest(self) -> Manifest:
-        # before we do anything else, patch the sources. This mutates
-        # results.disabled, so it needs to come before the final 'disabled'
-        # list is created
+        process_sources(self.manifest, project_name)
+        process_refs(self.manifest, project_name)
+        process_docs(self.manifest, self.root_project)
```

Comment:
If I understand it right, our next goal (in #3217) is to move these methods to _before_ the partial-parsing save point. Today, dbt fully re-processes every single source, ref, and description, every single time. Ideally, partial-parsing should only be re-processing nodes that have changed—whether the change occurred in a `.sql` file, `.yml` file, or `.md` file associated with that node.

Do I have that right?

Reply:
Yes, that's correct.